framework/contracts/native/ica-client/src/queries.rs

```diff
@@ -0,0 +1,278 @@
+use crate::state::{Config, CONFIG};
+use abstract_ica::{msg::ConfigResponse, ChainType, IcaAction, IcaActionResponse};
+use abstract_std::objects::TruncatedChainId;
+use cosmwasm_std::{ensure_eq, CosmosMsg, Deps, Env};
+
+use crate::{chain_types::evm, contract::IcaClientResult, error::IcaClientError};
+
+/// Timeout in seconds
+pub const PACKET_LIFETIME: u64 = 60 * 60;
+
+pub fn config(deps: Deps) -> IcaClientResult<ConfigResponse> {
+    let Config {
+        version_control,
+        ans_host,
+    } = CONFIG.load(deps.storage)?;
+    Ok(ConfigResponse {
+        ans_host: ans_host.address.to_string(),
+        version_control_address: version_control.address.into_string(),
+    })
+}
+
+pub(crate) fn ica_action(
+    deps: Deps,
+    env: Env,
+    _proxy_address: String,
+    chain: TruncatedChainId,
+    mut actions: Vec<IcaAction>,
+) -> IcaClientResult<IcaActionResponse> {
+    // match chain-id with cosmos or EVM
+    use abstract_ica::CastChainType;
+    let chain_type = chain.chain_type().ok_or(IcaClientError::NoChainType {
+        chain: chain.to_string(),
+    })?;
+
+    // todo: what do we do for msgs that contain both cosmos and EVM messages?
+    // Best to err if there's conflict.
+
+    // sort actions
+    // 1) Transfers
+    // 2) Calls
+    // 3) Queries
+    actions.sort_unstable();
```

Comment:
(see previous comment)
3) Fund transfers first, Execute second, Query last

I'm confused, which order should it be?

Beyond that, is it okay to mix order inside of the group? Won't it be better to create a new vector to which we push first group of actions, second group of actions, ...?

Reply:
I'm following this logic: https://github.com/srdtrk/cw-ica-controller/blob/44962d06a69852176b9417caa3326090af548aaf/src/types/msg.rs#L51

I.e. execute queries after calls, which makes sense to me because then you can easily request some post-action query data (if relevant). 

The problem is that some implementations will support this while others wont. For example, with cw-ica-controller we know that the actions and queries will be executed in that order, but for other solutions (like our current polytone implementation) we don't know if the execution will be performed before the queries because they are two different packets...

Reply:
I don't get the point of sorting it then at all

Reply:
Problem is that cw-ica-controller sorts them so we either need to do the same or we risk confusing devs.

Reply:
Removed the sorting for now.